Bottom: dc4f5b4a6a9aa43be9ee1914e329909e992e8a54
Top:    2c6671c69e9a56a6a8cec88eba71d1aac5b97c30
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2009-10-22 05:41:18 +0100

Workaround crash where contour is passed as NULL to poly_PreContour












---

diff --git a/src/polygon.c b/src/polygon.c
index 349886a..9bf6e83 100644
--- a/src/polygon.c
+++ b/src/polygon.c
@@ -237,6 +237,16 @@ original_poly (PolygonType * p)
       }
   }
   END_LOOP;
+  if (contour == NULL)
+    {
+      printf ("How did that escape - did the loop iterate zero times??\n");
+      POLYGONPOINT_LOOP (p);
+        {
+          printf ("Hello\n");
+        }
+      END_LOOP;
+      return NULL;
+    }
   poly_PreContour (contour, TRUE);
   /* make sure it is a positive contour */
   if ((contour->Flags.orient) != PLF_DIR)
